website/oauth/models/__init__.py

```diff
@@ -348,3 +355,68 @@ def handle_callback(self, response):
         :return dict:
         """
         pass
+
+
+class ApiOAuth2Application(StoredObject):
+    """Registration and key for user-created OAuth API applications"""
+    _id = fields.StringField(
+        primary=True,
+        default=lambda: str(ObjectId())
+    )
+
+    # Client ID and secret. Use separate ID field so ID format doesn't have to be restricted to database internals.
+    client_id = fields.StringField(default=lambda: uuid.uuid4().hex,  # Not *guaranteed* unique, but very unlikely
+                                   unique=True,
+                                   index=True)
+    client_secret = fields.StringField(default=generate_client_secret)
+
+    active = fields.BooleanField(default=True,  # Set to False if application is deactivated
```

Comment:
CAS checks this field name, and changing the field name in the OSF model would cause problems.

For now, I will make all the other changes, but am leaving this one deliberately untouched. It can be revisited separately when Michael is not backlogged.